### ISSUE TITLE
error setting scaling_configuration: Invalid address to set: []string{"scaling_configuration"}

### ISSUE THREAD

**Issue by rojoangel** [terraform-providers/terraform-provider-aws#5659]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
```
Terraform v0.11.8
+ provider.aws: version 1.33.0
```

### Affected Resource(s)

* aws_rds_cluster

### Terraform Configuration Files

```hcl
data "aws_rds_cluster" "main_cluster" {
  cluster_identifier = "main-cluster"
}
```

### Expected Behavior

Terraform should have refreshed state of cluster and continue with planing.

### Actual Behavior

```
Error: Error refreshing state: 1 error(s) occurred:

* module.registry-service.data.aws_rds_cluster.main_cluster: 1 error(s) occurred:

2018-08-23T15:24:37.912+0200 [DEBUG] plugin.terraform-provider-aws_v1.33.0_x4: 2018/08/23 15:24:37 [ERR] plugin: plugin server: accept unix /var/folders/93/8f2g60rx7xscfs7w2rp2mft40000gn/T/plugin247577689: use of closed network connection
* module.registry-service.data.aws_rds_cluster.main_cluster: data.aws_rds_cluster.main_cluster: error setting scaling_configuration: Invalid address to set: []string{"scaling_configuration"}
```

### Steps to Reproduce

1. `terraform init`
2. `terraform apply`

### Important Factoids

Provider updated from from 1.32.0 to 1.33.0. Configuration works with the older version.

### References

Thanks

**Comment by bflad** [comment]
Hi @rojoangel 👋 Really sorry for the hassle this has caused. For now, the workaround is to stay pinned on version 1.32.0 of the AWS provider when using that data source until this can be fixed.

We've seen regressions like this in the past with the `aws_rds_cluster` datasource (#4922) because it was sharing some of the code for reading RDS cluster information with the `aws_rds_cluster` resource, but the datasource schema was not updated at the same time. In that regard, I have submitted #5660 to fix the regression as well as prevent similar ones in the future by completely separating their logic.

**Comment by rojoangel** [comment]
@bflad kudos for the quick reaction! 👏 
Yes, for now we have fixed the version to 1.32.0 - that'll work for us until this gets fixed

**Comment by bflad** [comment]
The fix for this has been merged into master and will release with version 1.34.0 of the AWS provider, likely later today or tomorrow.

**Comment by bflad** [comment]
This has been released in [version 1.34.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1340-august-30-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
